text/0000-cargo-libstd-awareness.md

```diff
@@ -0,0 +1,245 @@
+- Feature Name: cargo_stdlib_awareness
+- Start Date: 2015-05-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+
+# Summary
+
+Currently, Cargo doesn't know what parts of the standard library packages depend on.
+By giving it this knowledge, we can make cross compilation and exotic platform development easier, simplify rustbuild, and allow anyone to easily specify bounds on the standard library for their packages.
+This will allow building parts of the standard library from source, but in order to not disrupt existing workflows, the binaries that come with rustc will still be used by default.
+
+
+# Motivation
+
+First, some background.
+`rustc` can be given crates and their location with `--extern <name>=<path>`.
+When finding a library, `rustc` first sees if its location has been specified with `--extern`, then looks in any directories specified with `-L`, and finally looks in the "sysroot" [specifically `<sysroot>/lib/rustlib/<target>/lib`].
+Cargo passes in all dependencies it builds with `--extern`.
+However Cargo does not know about the standard library, so builds of it are taken from the sysroot.
+
+For cross-compiling, one can often download standard library binaries with rustup.
+This is convenient, but one cannot expect pre-built binaries for all platforms.
+In particular, embedded systems often have detailed configurations to convey as much information as possible about the hardware to the compiler.
+Furthermore, not all of the stdlib is available on every platform---there is an RFC in the works to pre-build a smaller set of crates for [Cortex-M microcontrollers](https://github.com/rust-lang/rfcs/pull/1645).
+It would be nice to know if the available subset is adequate before attempting a build.
+We can only do that if all packages have explicit standard library deps to cross-reference with platform requirements of each standard library crate.
+
+Now rustup could be augmented to build stdlib binaries in addition to downloading them, but we have extra configuration options for the standard library such as [panic strategies](https://github.com/rust-lang/rfcs/blob/master/text/1513-less-unwinding.md) along with plans to add more such as [`core` without floats](https://github.com/rust-lang/rfcs/issues/1364).
```

Comment:
It appears that "core without floats", in the sense of a Cargo feature that `cfg`s away all float related code from `core`, is no longer required. Tweaking LLVM features appears to fix all the problems that x86 kernel devs were having. I'm waiting for [confirmation on this matter](https://github.com/rust-lang/rfcs/issues/1364#issuecomment-236374104).